### PR TITLE
feat(blocks) Add ollama integration

### PR DESCRIPTION
**Pull request by aarushik93**
### Background

Add ollama for local model support

### Changes 🏗️

Added ollama provider and call

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
Enhancement

___

### **Description**
- Integrated Ollama as a new LLM provider in the `llm_call` function.
- Added `OLLAMA_LLAMA3_8B` model to the `LlmModel` enum.
- Included `ollama_api_key` in the `LlmApiKeys` dictionary.
- Updated the `Secrets` class to include `ollama_api_key`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>llm.py</strong><dd><code>Integrate Ollama LLM Provider in LLM Call Block</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/blocks/llm.py

<li>Added import for <code>ollama</code><br> <li> Included <code>ollama_api_key</code> in <code>LlmApiKeys</code><br> <li> Added <code>OLLAMA_LLAMA3_8B</code> to <code>LlmModel</code> enum<br> <li> Integrated <code>ollama</code> in <code>llm_call</code> function<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7505/files#diff-a3667eee15cd39a1cf134cc1d251e88c26cae672ae0de2cac1988e101ba095df">+11/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>settings.py</strong><dd><code>Add Ollama API Key to Secrets Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/util/settings.py

- Added `ollama_api_key` field to `Secrets` class

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7505/files#diff-281666fe9783cb96474f35739db160eceb136aca260e20e73990edc91f19610a">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/util/settings.py [hunk 1]
```diff
@@ -74,6 +74,7 @@ class Secrets(UpdateTrackingModel["Secrets"], BaseSettings):
     openai_api_key: str = Field(default="", description="OpenAI API key")
     anthropic_api_key: str = Field(default="", description="Anthropic API key")
     groq_api_key: str = Field(default="", description="Groq API key")
+    ollama_api_key: str = Field(default="", description="Ollama API key")
```
Comment (aarushik93) [inline rnd/autogpt_server/autogpt_server/util/settings.py]: this is not needed, but the validation complains on this. Alternatively i can create a "LocalLLMCall" block

**Review by majdyz — APPROVED** [review]
Make sure you run `poetry run format` before merging to make everything green.

File: rnd/autogpt_server/autogpt_server/blocks/llm.py [hunk 2]
```diff
@@ -16,6 +17,7 @@
     "openai": BlockSecret("openai_api_key"),
     "anthropic": BlockSecret("anthropic_api_key"),
     "groq": BlockSecret("groq_api_key"),
+    "ollama": BlockSecret("ollama_api_key"),
```
Comment (majdyz) [inline rnd/autogpt_server/autogpt_server/blocks/llm.py]: You can use BlockSecret(value="") instead and remove everything about ollama_api_key

Comment (aarushik93) [reply rnd/autogpt_server/autogpt_server/blocks/llm.py]: oh sweet, thanks will do that

File: rnd/autogpt_server/autogpt_server/util/settings.py (hunk 1 above)
Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/util/settings.py]: https://github.com/Significant-Gravitas/AutoGPT/pull/7505/files#r1684171354

**Review by majdyz — APPROVED** [review]

**Pull request merged**